activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -273,17 +273,15 @@ def load_target
         if find_target?
           @target = merge_target_lists(find_target, target)
         elsif target.empty? && set_through_target_for_new_record?
-          reflections = reflection.chain
-          reflections.pop
-          reflections.reverse!
+          reflections = reflection.chain.reverse!
 
-          @target = reflections.reduce(through_association.target) do |middle_target, reflection|
-            if middle_target.empty?
+          @target = reflections.each_cons(2).reduce(through_association.target) do |middle_target, (middle_reflection, through_reflection)|
+            if middle_target.nil? || (middle_reflection.collection? && middle_target.empty?)
```

Comment:
hi @joshuay03 I got this error on `middle_reflection` : 
```ruby
 undefined method 'collection?' for an instance of ActiveRecord::Reflection::PolymorphicReflection 
 ```
 
 when doing this : `FactoryBot.create(:perk_service, banking_actors: [element] )`
 
```ruby
  has_many :perk_servicings, dependent: :destroy
  has_many :banking_actors, through: :perk_servicings, source: :perk_servicable, source_type: "BankingActor"
```

worth noting that banking_actors, is a through as noted above. and `PerkServicing` is a polymorphic join table ( with `belongs_to :perk_serviceable, polymorphic: true` )

Reply:
Thanks, I'll have a look.

Reply:
https://github.com/rails/rails/pull/54046